File: csharp/secret-handshake/SecretHandshake.cs
public static class SecretHandshake
{
    private static readonly string[] signals = [
        "wink",
        "double blink",
        "close your eyes",
        "jump",
    ];

    public static string[] Commands(int commandValue)
    {
        var results = new List<string>();

        for (var i = 0; i < signals.Length; i++)
        {
            if ((commandValue & ((int)Math.Pow(2, i))) != 0)
            {
                results.Add(signals[i]);
            }
        }

        if ((commandValue & 16) != 0)
        {
            results.Reverse();
        }

        return [.. results];
    }
}

File: cpp/secret-handshake/secret_handshake.cpp
#include "secret_handshake.h"
#include <array>
#include <cmath>
#include <algorithm>

namespace secret_handshake
{

    static const std::array<std::string, 4> signals{
        "wink",
        "double blink",
        "close your eyes",
        "jump"};

    std::vector<std::string> commands(int command_value)
    {
        std::vector<std::string> results;

        for (size_t i = 0; i < signals.size(); i++)
        {
            if ((command_value & ((int)pow(2, i))) != 0)
            {
                results.push_back(signals[i]);
            }
        }

        if ((command_value & 16) != 0)
        {
            std::reverse(results.begin(), results.end());
        }

        return results;
    }

} // namespace secret_handshake
